### PR TITLE
ofdpa-api: update stp data model

### PR DESCRIPTION
**Pull request by rubensfig**

### CODE REVIEW

**Comment by hilmarm** [comment]
lgtm

**Comment by KanjiMonster** [comment]
So is ofdpa-grpc supposed to handle the VID -> STG mapping? There are only 511 STGs available, so you need to keep track of which VIDs you have assigned to which STGs in ofdpa-grpc so you translate them correctly for the bcm-command.

Also how do you then set the global state? setting vlan to 0?

**Comment by hilmarm** [comment]
> So is ofdpa-grpc supposed to handle the VID -> STG mapping? There are only 511 STGs available, so you need to keep track of which VIDs you have assigned to which STGs in ofdpa-grpc so you translate them correctly for the bcm-command.
> 
> Also how do you then set the global state? setting vlan to 0?

Good catch, for now we will map VID to STG with the identity function (VID 1 -> STG 1), so we will be restricted to vlans in the range 511. But we should still change the interface for the VlanAdd function to take a VID and STG (currently the function simply adds VID <x> to STG <x>), so in future implementations we can create a mapping somewhere. (store state in basebox?)

The global state is handled by a different function, ofdpaStpStatePortSet. (Note difference in ..StgState.. and StpState.. , stg sets per vlan with the stg function in ofdpa-api, stp calls the port function in ofdpa-api bcm)

Passing VID=0 to ofdpaStgStatePortSet will set the port state in the STG group 1. This was introduced to maintain some backwards compatibility.

File: api/ofdpa.proto [hunk 1]
```diff
@@ -1,7 +1,5 @@
 syntax = "proto3";
-
 package ofdpa;
-import "stp/openconfig-spanning-tree.proto";
```
Comment (jklare) [inline api/ofdpa.proto:4]: would it be possible to extend/patch the already existing proto instead of dropping it and reinventing the wheel?

Comment (hilmarm) [reply api/ofdpa.proto:4]: The problem with the data model was that the data structure for PerVlan STP (PVST) was very large, and we only needed two or three variables from it. This turns out to be enough for the STP calls as well (only port name and state needed)

The Openconfig files had a PVST variable, containing list of VLANs, which again contain vlan id in one variable, and another list of interfaces again.

**Pull request merged**